### PR TITLE
Protect against XSS attacks

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
- Adds regex patterns to API spec to be more selective about accepted input using Confluence page _ATAT Business Rules (Data Fields)_ as guidance
- Adds package `xss`
- Adds custom [middy](https://github.com/middyjs/middy) middleware `xssSanitizer`
- Adds simple unit test for `xssSanitizer`
- `xssSanitizer` is used in `createPortfolioStep`
- Add new XSS unit tests for `createPortfolioStep`

TODO:

- [x] correct failing unit tests
- [ ] test in a deployed environment

Builds on #302

Ticket: AT-6615

### CODE REVIEW

File: atat_provisioning_wizard_api.yaml [hunk 1]
```diff
@@ -977,7 +983,9 @@ components:
       additionalProperties: false
       properties:
         task_order_number:
+          description: TO numbers are 13 characters. TO modifications are 17 characters.
           type: string
+          pattern: '^[\d]{13,17}$'
```
Comment (juliusfitzhugh-ccpo) [inline atat_provisioning_wizard_api.yaml:988]: Not sure if the full business rule is in scope for this ticket, but this pattern would probably need to modified in the future to meet the full business rule for `Task Order Number`.

**Review by juliusfitzhugh-ccpo — COMMENTED** [review]
Looks good so far. Small comment about the `Task Order Number`.

File: atat_provisioning_wizard_api.yaml (hunk 1 above)
Comment (tomarnold-ccpo) [reply atat_provisioning_wizard_api.yaml:988]: @juliusfitzhugh-ccpo these were changed to be more specific just a few days ago. We still haven't had the chance to update the ui with these new requirements.  Imo this is sufficient for now, and we can create another ticket to capture the additional business rules

File: packages/api/portfolioDrafts/xssSanitizer.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+// keep plain text only
+const xssOptions = {
+  whiteList: {}, // empty means remove all tags
+  stripIgnoreTag: true, // remove all tags not in whitelist
+  stripIgnoreTagBody: ["script"], // remove script tag content
+};
+const xssSanitizer = (): middy.MiddlewareObj<APIGatewayProxyEvent, APIGatewayProxyResult> => {
+  const before: middy.MiddlewareFn<APIGatewayProxyEvent, APIGatewayProxyResult> = async (request): Promise<void> => {
+    request.event.body = JSON.parse(xss(JSON.stringify(request.event.body ?? {}), xssOptions));
```
Comment (kylelaker-ccpo) [inline packages/api/portfolioDrafts/xssSanitizer.ts:13]: Odd that this is coming in as an object already, but this does seem to be necessary. Without the JSON stringify/parse `xss` just gives the string `[object Object]`.

Comment (jasonburkert-ccpo) [reply packages/api/portfolioDrafts/xssSanitizer.ts:13]: Thanks for the help.

**Pull request merged**